library/alloc/src/vec/in_place_collect.rs

```diff
@@ -186,6 +186,26 @@ const fn in_place_collectible<DEST, SRC>(
     }
 }
 
+const fn needs_realloc<SRC, DEST>(src_cap: usize, dst_cap: usize) -> bool {
+    if const { mem::align_of::<SRC>() != mem::align_of::<DEST>() } {
+        return src_cap > 0;
+    }
+
+    // If src size is an integer multiple of the destination size
+    // the resulting capacity will also always be an integer multiple without remainder
```

Comment:
```suggestion
    // If src size is an integer multiple of the destination size, we know the capacity we calculate
    // with src_cap * size_of::<SRC> / size_of::<DEST> does not have a remainder.
```
It took me a long time to understand this comment. It's referring to the division done outside this function. I've tried to make that clear.

r=me with this suggestion or equivalent clarification